src/lib/accessibility/FocusZone/FocusZone.tsx

```diff
@@ -0,0 +1,950 @@
+      this.updateTabIndexes()
+
+      if (this.props.defaultActiveElement) {
+        this._activeElement = document.querySelector(this.props.defaultActiveElement) as HTMLElement
+        this.focus()
+      }
+    }
+  }
+
+  public componentWillUnmount() {
+    delete _allInstances[this._id]
+    if (this.windowElement) {
+      this.windowElement.removeEventListener('keydown', this.onKeyDownCapture, true)
+    }
+  }
+
+  render() {
+    const { className } = this.props
+    const Tag = this.props.as || 'div'
+    const rest = getUnhandledProps({ handledProps: FocusZone.handledProps }, this.props)
+
+    return (
+      <Tag
+        role="presentation"
+        {...rest}
+        className={cx(FocusZone.className, className)}
+        ref={elem => (this._root.current = ReactDOM.findDOMNode(elem) as HTMLElement)} // findDOMNode needed to get correct DOM ref with react-hot-loader, see https://github.com/gaearon/react-hot-loader/issues/964
+        data-focuszone-id={this._id}
+        onKeyDown={this._onKeyDown}
+        onFocus={this._onFocus}
+        onMouseDownCapture={this._onMouseDown}
+      >
+        {this.props.children}
+      </Tag>
+    )
+  }
+
+  /**
+   * Sets focus to the first tabbable item in the zone.
+   * @param {boolean} forceIntoFirstElement If true, focus will be forced into the first element, even if focus is already in the focus zone.
+   * @returns True if focus could be set to an active element, false if no operation was taken.
+   */
+  public focus(forceIntoFirstElement: boolean = false): boolean {
+    if (this._root.current) {
+      if (
+        !forceIntoFirstElement &&
+        this._root.current.getAttribute(IS_FOCUSABLE_ATTRIBUTE) === 'true' &&
+        this._isInnerZone
+      ) {
+        const ownerZoneElement = this.getOwnerZone(this._root.current) as HTMLElement
+
+        if (ownerZoneElement !== this._root.current) {
+          const ownerZone =
+            _allInstances[ownerZoneElement.getAttribute(FOCUSZONE_ID_ATTRIBUTE) as string]
+
+          return !!ownerZone && ownerZone.focusElement(this._root.current)
+        }
+
+        return false
+      }
+      if (
+        !forceIntoFirstElement &&
+        this._activeElement &&
+        this._root.current.contains(this._activeElement) &&
+        isElementTabbable(this._activeElement)
+      ) {
+        this._activeElement.focus()
+        return true
+      }
+
+      const firstChild = this._root.current.firstChild as HTMLElement
+
+      return this.focusElement(getNextElement(this._root.current, firstChild, true) as HTMLElement)
+    }
+    return false
+  }
+
+  /**
+   * Sets focus to the last tabbable item in the zone.
+   * @param {boolean} forceIntoFirstElement If true, focus will be forced into the first element, even if focus is already in the focus zone.
+   * @returns True if focus could be set to an active element, false if no operation was taken.
+   */
+  public focusLast(): boolean {
```

Comment:
What is the purpose of this method? It is not called anywhere.

Reply:
It is a public API. There is a wrong comment which describes an argument which does not exist. I fixed the comment but we agreed with Juraj that we will keep this method in the API.

Reply:
I just noticed that this method is missing from the interface so I added it there. :-)